### DISCUSSION TITLE
Testing redirect with Japa's api client is failing

### DISCUSSION THREAD

**Discussion by arandilopez** [adonisjs/core#3733] [Help] [answered]
Hey everyone! I started a small project a few days ago and I'm being blocked with this issue that I replicated [in this repo](https://github.com/arandilopez/redirect-test).

As you can see [the test](https://github.com/arandilopez/redirect-test/blob/master/tests/functional/redirect.spec.ts) is pretty simple, just calling an [enpoint](https://github.com/arandilopez/redirect-test/blob/master/start/routes.ts#L27) that redirects to another. The issue I'm seeing in the tests is this:
```sh
[ info ]  running tests...
❯ @japa/run-failed-tests: Running only 1 failed test(s)
{"level":30,"time":1652989564243,"pid":306395,"hostname":"pop-os","name":"redirect-test","msg":"started server on 0.0.0.0:38333"}

functional / Redirect (tests/functional/redirect.spec.ts)
  ✖ redirect works as expected (31ms)

 FAILED 

total        : 1
failed       : 1
duration     : 37ms


✖ redirect works as expected

   connect ECONNREFUSED 0.0.0.0:80
  
   at TCPConnectWrap.afterConnect [as oncomplete] net.js:1148
```

As you can see the client tries to request at port 80 when the TestUtils' server is starting it at 38333. I added a nock to mock `0.0.0.0:80` and make the test pass to inspect the response's redirects with `console.log(response.redirects())` and getting `['0.0.0.0:/']`.

```sh
[ info ]  running tests...
{"level":30,"time":1652990523948,"pid":323119,"hostname":"pop-os","name":"redirect-test","msg":"started server on 0.0.0.0:41419"}

functional / Redirect (tests/functional/redirect.spec.ts)
[ '0.0.0.0:/' ]
  ✔ redirect works as expected (42ms)

 PASSED 

total        : 1
passed       : 1
duration     : 49ms
```

I don't know if this is an issue of the Response/Redirect behavior in test environtment or maybe the ApiClient has the issue with the redirect.

**Comment by thetutlage** [comment, marked as answer]
Fixed in https://github.com/japa/api-client/commit/9b2012025cd9941ec17206713fbb454a06ae9839

**Reply by arandilopez** [reply]
Amazing! Thanks
